modPrecompiled/src/org/aion/precompiled/contracts/TXHashContract.java

```diff
@@ -0,0 +1,30 @@
+package org.aion.precompiled.contracts;
+
+import javax.annotation.Nonnull;
+import org.aion.base.type.IExecutionResult;
+import org.aion.vm.AbstractExecutionResult.ResultCode;
+import org.aion.vm.ExecutionContext;
+import org.aion.vm.ExecutionResult;
+import org.aion.vm.IPrecompiledContract;
+
+public class TXHashContract implements IPrecompiledContract {
+    public static final long COST = 500L;
+
+    private final ExecutionContext context;
+
+    public TXHashContract(@Nonnull final ExecutionContext _context) {
```

Comment:
there's no actual standard for non-null annotations: https://stackoverflow.com/questions/35892063/which-nonnull-java-annotation-to-use; I think null checks would be a better choice here

Reply:
the context should always non-null when constructing the Class. Adding this mainly tell the compiler the input should non-null.